src/components/Onboarding/OnboardingScreen.js

```diff
@@ -0,0 +1,668 @@
+}
+
+function Step3({
+  currentStep, formState, classes, role, handleChange, onPrev, handleSubmit, errorMsg,
+  step3Check, step3IsInvalid, loading,
+}) {
+  const COMPANY_NAMES = ['Company Name', 'Agency', 'Farm Name'];
+  if (currentStep !== 3) {
+    return null;
+  }
+  return [
+    <div className={classes.subtitleText}>
+      Contact Information
+    </div>,
+    <Grid
+      container
+      spacing={2}
+      alignItems="center"
+    >
+      <Grid item xs={12}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          name="org"
+          label={COMPANY_NAMES[role]}
+          value={formState.org}
+          onChange={handleChange}
+        />
+      </Grid>
+      <Grid item xs={12}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          label="Location"
+          name="zipcode"
+          value={formState.zipcode}
+          onChange={handleChange}
+        />
+      </Grid>
+      <Grid item xs={12} sm={6}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          id="fname"
+          label="First Name"
+          name="firstName"
+          placeholder="First Name"
+          value={formState.firstName}
+          onChange={handleChange}
+        />
+      </Grid>
+      <Grid item xs={12} sm={6}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          id="lname"
+          name="lastName"
+          placeholder="Last Name"
+          value={formState.lastName}
+          onChange={handleChange}
+        />
+      </Grid>
+      <Grid item xs={12}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          label="Phone"
+          name="phone"
+          value={formState.phone}
+          onChange={handleChange}
+        />
+      </Grid>
+      <Grid item xs={12}>
+        <TextField
+          variant="outlined"
+          required
+          fullWidth
+          label="Comments"
+          name="comments"
+          value={formState.comments}
+          onChange={handleChange}
+        />
+      </Grid>
+      {errorMsg && <Grid item xs={12} className="error-msg">{errorMsg}</Grid>}
+      {step3Check && <Grid item xs={12} className="error-msg">{step3Check}</Grid>}
+      <Grid item xs={12} sm={6}>
+        <Button
+          type="button"
+          onClick={onPrev}
+          fullWidth
+          color="primary"
+          variant="contained"
+          className={classes.blackButton}
+        >
+          Back
+        </Button>
+      </Grid>
+      <Grid item xs={12} sm={6}>
+        <Button
+          className={classes.greenButton}
+          type="button"
+          onClick={handleSubmit}
+          fullWidth
+          color="primary"
+          variant="contained"
+          disabled={step3IsInvalid || loading}
+        >
+          Finish
+        </Button>
+      </Grid>
+    </Grid>,
+  ];
+}
```

Comment:
I know this is a lot of comments, so this can definitely be a stretch goal, but can you refactor your code so that all of your Steps are in their own files? That would organize the code a lot!